iroha/src/lib.rs

```diff
@@ -268,21 +281,24 @@ impl Iroha {
         });
 
         let queue = Arc::clone(&self.queue);
-        if let Some(ref genesis_block) = self.genesis_block {
-            log::info!("Initialize iroha using the genesis block");
-            for transaction in genesis_block.transactions.iter() {
-                if let Err(e) = queue
-                    .write()
-                    .await
-                    .push_pending_transaction(transaction.clone())
-                {
-                    log::error!(
-                        "Failed to push transactions from genesis block to queue error: {}",
-                        e
-                    )
+        let sumeragi = Arc::clone(&self.sumeragi);
+        self.waiting_for_peers_are_active().await.map(|_| async {
```

Comment:
You can do like that to get the peer set to my mind:
`self.wait_for_active_peers(&sumeragi.read().await.network_topology.sorted_peers)`

This way you would not need to move out `NetworkTopology` struct anywhere.